packages/block-directory/src/plugins/index.js

```diff
@@ -2,14 +2,29 @@
  * WordPress dependencies
  */
 import { registerPlugin } from '@wordpress/plugins';
+import { addFilter } from '@wordpress/hooks';
 
 /**
  * Internal dependencies
  */
 import InserterMenuDownloadableBlocksPanel from './inserter-menu-downloadable-blocks-panel';
+import missingInstallView from './missing-install-view';
 
 registerPlugin( 'block-directory', {
 	render() {
 		return <InserterMenuDownloadableBlocksPanel />;
 	},
 } );
+
+addFilter(
+	'blocks.registerBlockType',
+	'block-directory/fallback',
+	( settings, name ) => {
+		if ( name !== 'core/missing' ) {
+			return settings;
+		}
+		settings.edit = missingInstallView;
```

Comment:
It's fine as a temporary approach. However, it could be very hard to maintain in the long run as you would have to keep the implementation in sync with changes applied to the Missing block.

Have you considered using an approach where you could parameterize action items in some way? It could be also done through filters but on the more granular level or through SlotFill approach.